### PR TITLE
Add Recorded Future Identity Solution.

### PR DESCRIPTION
**Pull request by recordedfuture-dev**
Change(s):
   - Add Recorded Future Identity Solution, consist of 5 playbooks:
      - "RecordedFutureIdentity-search-workforce-user" playbook - to search new exposures for Workforce users
      - "RecordedFutureIdentity-search-external-user" playbook - to search new exposures for External users
      - "RecordedFutureIdentity-add-AAD-security-group-user" playbook - to add risky user to Active Directory Security Group for users at risk
      - "RecordedFutureIdentity-confirm-AAD-risky-user" playbook - to confirm to Active Directory Identity Protection that user is compromised
      - "RecordedFutureIdentity-lookup-and-save-user" playbook - to lookup additional information on a compromised user and save results to Log Analytics

   Reason for Change(s):
   - This Solution shows use of new Recorded Future Identity API via [Recorded Future Identity Connector](https://docs.microsoft.com/en-us/connectors/recordedfutureidenti/).

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Review by NikTripathi — APPROVED** [review]
Approving package. Can merge once playbooks are approved.

**Comment by NikTripathi** [comment]
@anki-narravula Can you please approve playbooks.

**Review by anki-narravula — CHANGES_REQUESTED** [review]
Hi @recordedfuture-dev Sorry for delay in responding. We were finalizing the new specs for playbooks, so got delayed.  Please find my feedback below

1. We have recently changed the solutioning approach. Can you please use V2 version of tool to create package, [new version can be found here]( https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2) . All packages that are created using V2 version of the tool starts with v2.0.0 so, please use the version for your package 2.0.0
2. With the new version, all the playbooks install as template gallery instead of active resources. So, for all the playbooks metadata object with all the details (relevant) mentioned is mandatory. Please [refer the guidelines here](https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide#contribution-guidelines). You also can refer some of the [playbooks like this](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Okta%20Single%20Sign-On/Playbooks/OktaPlaybooks/Okta-EnrichIncidentWithUserDetails/azuredeploy.json) for an example
3. Since some playbooks are dependent on others, if not mentioned already n readme file, please mention clearly on order of deployment.

**Comment by v-mchatla** [comment]
Hi @recordedfuture-dev 
Can you please incorporate the changes requested by @anki-narravula 
Thanks

**Comment by RecordedFutureOskbo** [comment]
We now have a mainTemplate.json that validates by `Test-AzTemplate`. But we run in to a issue when testing using Custom deployment. 

Deployment template validation failed: 'The resource 'Microsoft.Resources/templateSpecs/OskarFreeRF-pl-b4mxdohwsqsq4' at line '82' and column '9' is defined multiple times in a template.

Do you have any advise?

**Review by RecordedFutureOskbo — DISMISSED** [review]

**Comment by NikTripathi** [comment]
@v-mchatla Please look into this.

**Comment by RecordedFutureOskbo** [comment]
@NikTripathi We now have a Working solution

File: Solutions/Recorded Future Identity/Data/Solution_RecordedFutureIdentity.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "Analytic Rules": [],
```
Comment (v-mchatla) [inline Solutions/Recorded Future Identity/Data/Solution_RecordedFutureIdentity.json:6]: Please remove the analytic rule and workbook fields as we have only playbooks.

Comment (RecordedFutureOskbo) [reply Solutions/Recorded Future Identity/Data/Solution_RecordedFutureIdentity.json]: I'm using the createSolutionV2.ps1 script in the V2 directory and the input file contains 

``` json
  "BasePath": "c:\\github\\Azure-Sentinel\\Solutions\\Recorded Future Identity\\",
  "Version": "2.0.0",
  "Metadata": "SolutionMetadata.json",
```
But the mainTemplate still reports this

``` json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "metadata": {
    "author": "Recorded Future Premier Integrations - support@recordedfuture.com",
    "comments": "Solution template for Recorded Future Identity"
  },
```

Comment (recordedfuture-dev) [reply Solutions/Recorded Future Identity/Data/Solution_RecordedFutureIdentity.json]: We have made the changes regarding the package version, and removed analytic rule and workbook fields.

About "removing playbook description from data file as we will have it in Create UI file":
the CreateUI file is generated out of "Solution_RecordedFutureIdentity.json" file, and if we remove the playbook descriptions from Solution_RecordedFutureIdentity.json file - on a next version release, when CreateUI file will be re-generated from Solution_RecordedFutureIdentity.json file - we will "lose" the playbook descriptions (as they wouldn't be present in the Solution_RecordedFutureIdentity.json file, on which CreateUI file is based).

So I don't think we should remove playbook description from Solution_RecordedFutureIdentity.json file.

If you meant something else - please specify more details. Thanks!


Other than that - all changes are done and ready for your review.
Review once again, please!

**Review by anki-narravula — APPROVED** [review]
Playbooks looks good. Approved

**Review by v-mchatla — APPROVED** [review]
Changes looks good. Approving the PR

**Pull request merged**